### ISSUE TITLE
extras_require doesn't work with wildcard requirements

### ISSUE THREAD

**Issue by dabacon** [quantumlib/Cirq#1900]
In setup.py we now have extras_require to only install some dependencies when requested.  Like
```
pip install cirq[contrib]
```
However this appears to not work when we use this setup.py to create a wheel and upload it to pypi.  It yells about being unable to support a version when publishing the package.
```
HTTPError: 400 Client Error: Invalid value for requires_dist. Error: Invalid version: '1.3.*' for url: https://upload.pypi.org/legacy/
```

**Comment by cduck** [comment]
I was able to reproduce the error.  Somehow it works for this package: https://pypi.org/project/RelStorage/#files

**Comment by dabacon** [comment]
Fixed in #1935
